pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -106,14 +106,24 @@ func NewGarbageCollector(
 		sharedInformers:  sharedInformers,
 		ignoredResources: ignoredResources,
 	}
-	if err := gb.monitorsForResources(deletableResources); err != nil {
-		return nil, err
+	if err := gb.syncMonitors(deletableResources); err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to sync all monitors: %v", err))
 	}
 	gc.dependencyGraphBuilder = gb
 
 	return gc, nil
 }
 
+// SyncResourceMonitors starts or stops resource monitors as needed to ensure
+// that all (and only) those resources present in the map are monitored.
+func (gc *GarbageCollector) SyncResourceMonitors(deletableResources map[schema.GroupVersionResource]struct{}) error {
```

Comment:
Sorry i just saw the comment in my email. github hided the comment by default as the code is outdated.

> In the case of the missed RC delete event, doesn't that leave a virtual node for the RC which is never re-verified with an existence check? Even if there was a future update to the owned pod, the virtual node which owns the pod doesn't seem to be re-queued for a deletion attempt (correct me if I'm wrong).

You are right.

> I'm not sure that syncing informer checks would ever totally close this gap: it should be assumed that events can be missed.

I disagree. The informer won't miss any event after the initial list, that's a guarantee provided by etcd. So as long as the workers don't start before all the informers are synced, we won't hit that bug.

> Shouldn't the entire graph be periodically reconciled somehow to ensure consistency?

My concern is the performance implication. The informer framework gives us enough guarantees to avoid resyncing. 

> For example, at the least, periodically attempting to delete any virtual nodes that exists in order to account for the inevitable missed event.

If gc confirms the existence of the object via a GET, the node becomes real :)